service/controller/v1/resource/drainer/create.go

```diff
@@ -0,0 +1,159 @@
+
+			r.logger.LogCtx(ctx, "level", "debug", "message", "guest cluster node not found")
+			r.logger.LogCtx(ctx, "level", "debug", "message", "setting node config status of guest cluster node to final state")
+
+			drainerConfig.Status.Conditions = append(drainerConfig.Status.Conditions, drainerConfig.Status.NewFinalCondition())
+
+			_, err := r.g8sClient.CoreV1alpha1().DrainerConfigs(drainerConfig.GetNamespace()).UpdateStatus(&drainerConfig)
+			if err != nil {
+				return microerror.Mask(err)
+			}
+
+			r.logger.LogCtx(ctx, "level", "debug", "message", "set node config status of guest cluster node to final state")
+			resourcecanceledcontext.SetCanceled(ctx)
+			r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource for custom object")
+
+			return nil
+		} else if err != nil {
+			return microerror.Mask(err)
+		}
+
+		r.logger.LogCtx(ctx, "level", "debug", "message", "cordoned guest cluster node")
+	}
+
+	var customPods []v1.Pod
+	var systemPods []v1.Pod
+	{
+		r.logger.LogCtx(ctx, "level", "debug", "message", "looking for all pods running on the guest cluster node")
+
+		fieldSelector := fields.SelectorFromSet(fields.Set{
+			"spec.nodeName": key.NodeNameFromDrainerConfig(drainerConfig),
+		})
+		listOptions := apismetav1.ListOptions{
+			FieldSelector: fieldSelector.String(),
+		}
+		podList, err := k8sClient.CoreV1().Pods(v1.NamespaceAll).List(listOptions)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		for _, p := range podList.Items {
+			if p.GetNamespace() == "kube-system" {
+				systemPods = append(systemPods, p)
+			} else {
+				customPods = append(customPods, p)
+			}
+		}
+
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("found %d pods running custom workloads", len(customPods)))
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("found %d pods running system workloads", len(systemPods)))
+	}
+
+	if len(customPods) > 0 {
+		r.logger.LogCtx(ctx, "level", "debug", "message", "deleting all pods running custom workloads")
+
+		for _, p := range customPods {
+			err := k8sClient.CoreV1().Pods(p.GetNamespace()).Delete(p.GetName(), &apismetav1.DeleteOptions{})
+			if err != nil {
+				return microerror.Mask(err)
+			}
+		}
+
+		r.logger.LogCtx(ctx, "level", "debug", "message", "deleted all pods running custom workloads")
+	} else {
+		r.logger.LogCtx(ctx, "level", "debug", "message", "no pods to be deleted running custom workloads")
+	}
+
+	if len(systemPods) > 0 {
+		r.logger.LogCtx(ctx, "level", "debug", "message", "deleting all pods running system workloads")
+
+		for _, p := range systemPods {
+			err := k8sClient.CoreV1().Pods(p.GetNamespace()).Delete(p.GetName(), &apismetav1.DeleteOptions{})
+			if err != nil {
+				return microerror.Mask(err)
+			}
+		}
+
+		r.logger.LogCtx(ctx, "level", "debug", "message", "deleted all pods running system workloads")
+	} else {
+		r.logger.LogCtx(ctx, "level", "debug", "message", "no pods to be deleted running system workloads")
+	}
```

Comment:
Just wondering here. What happens if custom pod has long graceful period. Should we wait for all customPods tobe deleted before starting deleting systemPods?

Reply:
Again, copy/paste. This is already how it is. I did not make this up now. Once we get a problem and gain more experiences with draining and its issues we can adapt. I do not see what and how and why we should optimize for now.